src/core/Table/Table.tsx

```diff
@@ -231,202 +231,45 @@ export const Table = <
-      // Fix the parent group of the selection button to not be resizable
-      const selectionGroupHeader = headerGroups[0].headers[0];
-      selectionGroupHeader.canResize = false;
-    });
-  };
-
-  const onFilterHandler = (
-    newState: TableState<T>,
-    action: ActionType,
-    previousState: TableState<T>,
-    instance?: TableInstance<T>,
-  ) => {
-    const previousFilter = previousState.filters.find(
-      (f) => f.id === action.columnId,
-    );
-    if (previousFilter?.value != action.filterValue) {
-      const filters = newState.filters.map((f) => {
-        const column = instance?.allColumns.find((c) => c.id === f.id);
-        return {
-          id: f.id,
-          value: f.value,
-          fieldType: column?.fieldType ?? 'text',
-          filterType: column?.filter ?? 'text',
-        };
-      }) as TableFilterValue<T>[];
-      onFilter?.(filters, newState);
-    }
-  };
-
-  const onSelectHandler = (
-    newState: TableState<T>,
-    instance?: TableInstance<T>,
-  ) => {
-    if (!instance?.rows.length) {
-      onSelect?.([], newState);
-      return;
-    }
-
-    const selectedData: T[] = [];
-    const newSelectedRowIds = {} as Record<string, boolean>;
-    instance.initialRows.forEach((row) => {
-      if (newState.selectedRowIds[row.id] && !isRowDisabled?.(row.original)) {
-        newSelectedRowIds[row.id] = true;
-        selectedData.push(row.original);
-      }
-    });
-    newState.selectedRowIds = newSelectedRowIds;
-    onSelect?.(selectedData, newState);
-  };
-
-  const onExpandHandler = (
-    newState: TableState<T>,
-    instance?: TableInstance<T>,
-  ) => {
-    if (!instance?.rows.length) {
-      onExpand?.([], newState);
-      return;
-    }
-
-    const expandedData: T[] = [];
-    instance.rows.forEach((row) => {
-      if (newState.expanded[row.id]) {
-        expandedData.push(row.original);
-      }
-    });
-    onExpand?.(expandedData, newState);
-  };
-
-  const tableStateReducer = (
-    newState: TableState<T>,
-    action: ActionType,
-    previousState: TableState<T>,
-    instance?: TableInstance<T>,
-  ): TableState<T> => {
-    switch (action.type) {
-      case TableActions.toggleSortBy:
-        onSort?.(newState);
-        break;
-      case TableActions.setFilter:
-        onFilterHandler(newState, action, previousState, instance);
-        break;
-      case TableActions.toggleRowExpanded:
-      case TableActions.toggleAllRowsExpanded:
-        onExpandHandler(newState, instance);
-        break;
-      case singleRowSelectedAction: {
-        newState = {
-          ...newState,
-          selectedRowIds: { [action.id]: true } as Record<string, boolean>,
-        };
-      }
-      case TableActions.toggleRowSelected:
-      case TableActions.toggleAllRowsSelected:
-      case TableActions.toggleAllPageRowsSelected: {
-        onSelectHandler(newState, instance);
-        break;
+  const tableStateReducer = React.useCallback(
```

Comment:
I am not sure whether to move it to separate file or not. The only down side having it in separate file is that function will have a lot of props.

Reply:
I'd say leave it here.